zubhub_frontend/zubhub/src/components/upload_file/uploadFile.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react';
+import clsx from 'clsx';
+import ImageIcon from '@material-ui/icons/Image';
+import CustomButton from '../button/Button';
+import {
+  handleFileButtonClick,
+  handleImageFieldChange,
+} from './uploadFileScripts.js';
+import { Typography, FormControl, FormHelperText } from '@material-ui/core';
+import { getFieldAndIndex } from '../../assets/js/utils/scripts';
+
+function UploadFile(props) {
+  const {
+    id,
+    fileType,
+    uploadButtonLabel,
+    classes,
+    countFilesText,
+    multiple,
+    wraperState,
+  } = props;
+
+  const { field, index } = getFieldAndIndex(id);
+  const uploadFileRefs = {
+    fileInput: React.useRef(null),
+  };
+
+  const [state, setState] = useState({
+    media_upload: {
+      upload_dialog: false,
+      images_to_upload: [],
+      videos_to_upload: [],
+      upload_info: {},
+      upload_percent: 0,
+      uploaded_images_url: [],
+      uploaded_videos_url: [],
+    },
+  });
+
+  const handleSetUploadFileState = obj => {
+    if (obj) {
+      Promise.resolve(obj).then(obj => {
+        setState(state => ({ ...state, ...obj }));
```

Comment:
Please use lodash library, cloneDeep method.
If there are nested objects inside the state like
`const temp1 = {a: { b: 'c' }};`

if destructuring to temp2 like
`const temp2 = { ...temp1 }`

temp1 and temp2 are two different objects but temp1.a and temp2.a will be same references. If we change the valie in temp1.a, it reflects in temp2.a